app/models/label/LabelTable.scala

```diff
@@ -375,10 +376,34 @@ object LabelTable {
       "label_type_value" -> labelMetadata.labelTypeValue,
       "severity" -> labelMetadata.severity,
       "temporary" -> labelMetadata.temporary,
+      //this one is different because we're pulling from the tags and label_tags databases
+      "tags" -> getTagsFromLabelId(labelMetadata.labelId),
```

Comment:
Instead of calling this function here, you should change the definition of the `LabelMetadata` class to include a `List[String]`, and then modify the `retrieveSingleLabelMetadata` function to call `getTagsFromLabelId`. It is best for the `LabelMetadata` to actually include all the label metadata!

Also, the convention we are trying to use for comments is to start with a space and a capital letter, and end with a period, e.g., `// This one ... label_tags databases.` (very nitpicky, but just letting you know what the conventions are :)

Also `tag` and `label_tag` are tables, not databases :wink:

Reply:
First of all, I think that you should modify the function you wrote to just return a list of strings, not a single string that has already been formatted. I think the formatting should be done on the front-end. But taking a union of the queries isn't really what I was going for.

I guess I should just explain a bit more about how some things on the backend work. So take a look at the `retrieveSingleLabelMetadata` function. I want to give you an idea of what the object looks like during each part of the last line of that function.
1. After `selectQuery(labelId)`, you some sort of query object, I believe. Nothing that you can really work with directly.
2. After `selectQuery(labelId).list`, you now have a list of tuples where each tuple represents a row returned by the query. So in this particular case, the object would look like
    `[(Int, String, Float, ...), (Int, String, Float, ...), ...]`.
    Here, the Int is `lb1.label_id`, the String is `lb1.gsv_panorama_id`, and the Float is `lp.heading`, then I cut off the rest for brevity :wink: 
3. Now the `.map(label => LabelMetadata.tupled(label))` basically means that for each tuple in the list, the `LabelMetadata.tupled` is called on it. The `tupled` function just takes a tuple and turns it into that type of object, but only if the data types of the entries in the tuple match the fields of the object you are trying to make. For example, `LabelMetadata` has fields that are Int, String, Float, ..., so if you call `LabelMetadata.tupled((5, "blah", 5.4, ...))`, you will get a `LabelMetadata` object. This is just a shortcut to calling the constructor for the object, which would look like this: `LabelMetadata(5, "blah", 5.4, ...)`.
    
    But if you already have all the data in a tuple, it would be super annoying to take all the data _out_ of the tuple and put it into the constructor. If the `tupled` function wasn't there, this step would look like `.map(label => LabelMetadata(label._1, label._2, label._3, ...))` instead of the clean looking `.map(label => LabelMetadata.tupled(label))`.

Now, let's give the `LabelMetadata` object a list of strings that are the tags as a field. So the definition would be:
`case class LabelMetadata(labelId: Int, gsvPanoramaId: String, ..., tags: List[String])`.

Now, assuming you have modified the function you wrote to give a `List[String]` instead of a `String`, you could change that last line in `retrieveSingleLabelMetadata` to use the less elegant constructor, which would look like this:
`.map(label => LabelMetadata(label._1, label._2, ..., getTagsFromLabelId(label._1)))`

And finally, if there are multiple places where you would have to change the syntax to the less elegant one, just define a helper function that will do the messy constructor business, and call that function instead. E.g.,
`.map(label => myHelperFunc(label, getTagsFromLabelId(label._1)))`

Reply:
Woah. That makes a lot more sense. Thanks for helping me out :)